src/main/java/sky/storage/Storage.java

```diff
@@ -0,0 +1,136 @@
+package sky.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import sky.TaskList;
+import sky.exception.TextNoMeaningException;
+import sky.task.Deadline;
+import sky.task.Event;
+import sky.task.Task;
+import sky.task.ToDo;
+
+/**
+ * The Storage class deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Returns a list of tasks stored in the task file in user's hard disk.
+     *
+     * @return A list of tasks.
+     * @throws TextNoMeaningException If unable to read from file.
+     */
+    public List<Task> load() throws IOException {
+        File file = new File(this.filePath);
+        if (!file.exists()) {
+            return new ArrayList<>();
+        }
+        try {
+            List<Task> taskList = new ArrayList<>();
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNext()) {
+                String textLine = scanner.nextLine();
+                char taskType = textLine.charAt(1);
+                boolean isMarked = textLine.charAt(4) == 'X';
+                String description = textLine.substring(7);
+
+                if (taskType == 'T') {
+                    Task task = new ToDo(description);
+                    taskList.add(task);
+                } else if (taskType == 'D') {
+                    int indexOfBracket = -1;
+                    for (int i = 0; i < description.length(); i++) {
+                        if (description.charAt(i) == '(') {
+                            indexOfBracket = i;
+                        }
+                    }
+                    String taskDescription = description.substring(0, indexOfBracket - 1);
+                    String taskBy = description.substring(indexOfBracket + 5, description.length() - 1);
+                    Task task = new Deadline(taskDescription, taskBy);
+                    taskList.add(task);
+                } else if (taskType == 'E') {
+                    int indexOfBracket = -1;
+                    for (int i = 0; i < description.length(); i++) {
+                        if (description.charAt(i) == '(') {
+                            indexOfBracket = i;
+                        }
+                    }
```

Comment:
I noticed that this block of code follows the arrow-head style. Perhaps you can try to change your code to avoid this for better readability? e.g using guard clauses for happy paths

Reply:
You are right that this code could be too nested. Thanks for pointing this out :)